# Pendeteksian Multikolinearitas




In [ ]:
install.packages("faraway");
library(faraway);

also installing the dependencies 'processx', 'glue', 'rprojroot', 'diffobj', 'rematch2', 'brio', 'callr', 'cli', 'desc', 'ellipsis', 'lifecycle', 'pkgload', 'praise', 'ps', 'rlang', 'waldo', 'withr', 'testthat', 'minqa', 'nloptr', 'RcppEigen', 'lme4'




  There are binary versions available but the source versions are later:
             binary    source needs_compilation
processx      3.5.2     3.5.3              TRUE
glue          1.4.2     1.6.2              TRUE
rprojroot     2.0.2     2.0.3             FALSE
diffobj       0.3.4     0.3.5              TRUE
brio          1.1.2     1.1.3              TRUE
cli           2.5.0     3.2.0              TRUE
desc          1.3.0     1.4.1             FALSE
lifecycle     1.0.0     1.0.1             FALSE
pkgload       1.2.1     1.2.4             FALSE
ps            1.6.0     1.7.0              TRUE
rlang        0.4.11     1.0.2              TRUE
waldo         0.2.5     0.4.0             FALSE
withr         2.4.2     2.5.0             FALSE
testthat      3.0.2     3.1.3              TRUE
nloptr      1.2.2.2     2.0.0              TRUE
RcppEigen 0.3.3.9.1 0.3.3.9.2              TRUE
lme4         1.1-26    1.1-29              TRUE

  Binaries will be installed
package 'processx' successfully 

installing the source packages 'rprojroot', 'desc', 'lifecycle', 'pkgload', 'waldo', 'withr'



In [4]:
longley;

GNP.deflator GNP     Unemployed Armed.Forces Population Year Employed
1947  83.0        234.289 235.6      159.0        107.608    1947 60.323  
1948  88.5        259.426 232.5      145.6        108.632    1948 61.122  
1949  88.2        258.054 368.2      161.6        109.773    1949 60.171  
1950  89.5        284.599 335.1      165.0        110.929    1950 61.187  
1951  96.2        328.975 209.9      309.9        112.075    1951 63.221  
1952  98.1        346.999 193.2      359.4        113.270    1952 63.639  
1953  99.0        365.385 187.0      354.7        115.094    1953 64.989  
1954 100.0        363.112 357.8      335.0        116.219    1954 63.761  
1955 101.2        397.469 290.4      304.8        117.388    1955 66.019  
1956 104.6        419.180 282.2      285.7        118.734    1956 67.857  
1957 108.4        442.769 293.6      279.8        120.445    1957 68.169  
1958 110.8        444.546 468.1      263.7        121.950    1958 66.513  
1959 112.6        482.704 381.3      255.2        123.366    1959 68.655  
1960 114.2        502.601 393.1      251.4        125.368    1960 69.564  
1961 115.7        518.173 480.6      257.2        127.852    1961 69.331  
1962 116.9        554.894 400.7      282.7        130.081    1962 70.551

Employed as Y variable and the others variable as X

In [3]:
dim(longley);

[1] 16  7

In [ ]:
cor(longley);

GNP.deflator GNP       Unemployed Armed.Forces Population
GNP.deflator 1.0000000    0.9915892  0.6206334  0.4647442   0.9791634 
GNP          0.9915892    1.0000000  0.6042609  0.4464368   0.9910901 
Unemployed   0.6206334    0.6042609  1.0000000 -0.1774206   0.6865515 
Armed.Forces 0.4647442    0.4464368 -0.1774206  1.0000000   0.3644163 
Population   0.9791634    0.9910901  0.6865515  0.3644163   1.0000000 
Year         0.9911492    0.9952735  0.6682566  0.4172451   0.9939528 
Employed     0.9708985    0.9835516  0.5024981  0.4573074   0.9603906 
             Year      Employed 
GNP.deflator 0.9911492 0.9708985
GNP          0.9952735 0.9835516
Unemployed   0.6682566 0.5024981
Armed.Forces 0.4172451 0.4573074
Population   0.9939528 0.9603906
Year         1.0000000 0.9713295
Employed     0.9713295 1.0000000



Dari hasil di atas kita dapat mengetahui korelasi baik itu antar variabel bebas maupun masing-masing korelasi variabel bebas dengan variabel Y. Contohnya pada korelasi antara GNP dengan GNP.deflator yang besar yaitu 0.9915892.



In [1]:
g <- lm(Employed~., longley)

Membuat linear regression dengan menjadikan Variabel Employed sebagai Y dan Semua variabel lainnya menjadi Variabel X (Dalam Function di atas ditulis menggunakan "." untuk mengambil seluruh variabel selain Employed).

In [2]:
summary(g)


Call:
lm(formula = Employed ~ ., data = longley)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.41011 -0.15767 -0.02816  0.10155  0.45539 

Coefficients:
               Estimate Std. Error t value Pr(>|t|)    
(Intercept)  -3.482e+03  8.904e+02  -3.911 0.003560 ** 
GNP.deflator  1.506e-02  8.492e-02   0.177 0.863141    
GNP          -3.582e-02  3.349e-02  -1.070 0.312681    
Unemployed   -2.020e-02  4.884e-03  -4.136 0.002535 ** 
Armed.Forces -1.033e-02  2.143e-03  -4.822 0.000944 ***
Population   -5.110e-02  2.261e-01  -0.226 0.826212    
Year          1.829e+00  4.555e-01   4.016 0.003037 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3049 on 9 degrees of freedom
Multiple R-squared:  0.9955,	Adjusted R-squared:  0.9925 
F-statistic: 330.3 on 6 and 9 DF,  p-value: 4.984e-10


Dari hasil *summary* di atas dapat diketahui:



1.   Koefisien Determinasi 0.9955 menunjukan sangat tinggi artinya setidaknya terdapat 1 yang signifikan. 99,55% keragaman dari variabel X dapat dijelaskan oleh variabel Y.
2.   Berdasarkan Uji T/Uji Parsial dari hasil summary di atas, dengan hipotesis:
*   $H_0$ : Tidak ada pengaruh variabel $X_n$ terhadap Y
*   $H_1$ : Terdapat pengaruh variabel $X_n$ terhadap Y

dan Kriteria Uji : <br>
$\hat{\alpha} < \alpha$ dan $t_{hitung} > t_{tabel}$, maka tolak $H_0$.

Keputusan dan Kesimpulan:
<br>
notes: <br> $\alpha = 0.05$
*   Variabel yang memiliki signifikansi terbesar yaitu variabel Armed.Forces dengan nilai p-value paling kecil dibandingkan p-value variabel lain yaitu 0.000944. Hal ini juga dapat dilihat dari korelasi antar variabel dengan Armed.forces yang cenderung kecil (dapat dilihat di cor(longley)).
*   Sementara variabel yang tidak signifikan yaitu variabel GNP,  GNP.deflator, dan Population dengan ketiga nilai p-value yang lebih besar dari 0.05.

Dengan taraf signifikansi 5%, terdapat pengaruh variabel Unemployed, Armed.forces, dan Year terhadap Employed sementara tidak ada pengaruh dari variabel GNP.deflator, GNP, dan Population.

3.  Berdasarkan Uji F/Uji Simultan di summary tersebut dapat diketahui: <br>

*   $H_0$ : Tidak ada pengaruh variabel X secara simultan(bersama-sama) terhadap variabel Y.
*   $H_1$ : Terdapat pengaruh variabel X secara simultan(bersama-sama) terhadap variabel Y.

Dengan Kritera Uji: <br>
$\hat{\alpha} < \alpha$ dan $F_{hitung} > F_{tabel}$, maka tolak $H_0$ <br>

Keputusan dan Kesimpulan:

*   Dengan nilai n = 16 dan k = 6 (Dari dim(longley)) maka $F_{tabel} = F_{\alpha, k,n-k} = F_{0.05,6,16-6} = F_{0.05,6,10} = 3.217175 $ (Dengan excel dapat menggunakan formula FINV(0.05,6,10).
*   List item















In [ ]:
summary(olsreg);

ERROR: ignored

In [ ]:
anova(olsreg);

Df  Sum Sq      Mean Sq     F value    Pr(>F)      
crim        1  6440.78306  6440.78306 286.030046 6.454140e-51
zn          1  3554.33620  3554.33620 157.845240 1.356741e-31
indus       1  2551.23644  2551.23644 113.298378 5.953713e-24
chas        1  1529.84788  1529.84788  67.939326 1.542886e-15
nox         1    76.24765    76.24765   3.386097 6.635054e-02
rm          1 10938.11661 10938.11661 485.753048 2.156727e-75
age         1    90.26791    90.26791   4.008726 4.581372e-02
dis         1  1779.50111  1779.50111  79.026227 1.163840e-17
rad         1    34.13432    34.13432   1.515878 2.188325e-01
tax         1   329.55407   329.55407  14.635234 1.472406e-04
ptratio     1  1309.30934  1309.30934  58.145385 1.266278e-13
black       1   593.33756   593.33756  26.349649 4.109280e-07
lstat       1  2410.83869  2410.83869 107.063426 7.776912e-23
Residuals 492 11078.78458    22.51785         NA           NA